### ISSUE TITLE
Delete postges pod after migrating to external DB

### ISSUE THREAD

**Issue by AlekseiSaff** [ansible/tower-operator#946]
##### ISSUE TYPE
 - postgres pod is creating again after migrating to external DB

##### SUMMARY
postgres pod is creating again after migrating to external DB

##### ENVIRONMENT
* AWX version: 22.1.0
* Operator version: 0.22.0
* Kubernetes version: minikube
* AWX install method: openshift, minishift, docker on linux, docker for mac, boot2docker

##### STEPS TO REPRODUCE

Migrate from awx-operator DB to external DB.

##### EXPECTED RESULTS

postgres POD needs to be removed

##### ACTUAL RESULTS

postgres pod is still available

##### ADDITIONAL INFORMATION

none

**Comment by rooftopcellist** [comment]
@AlekseiSaff Does your postgresql configuration secret have `type: unmanaged` in it?  

Since you are now using an external DB, it is not managed by the awx-operator.  Without the unmanaged type specified in the pg secret, the operator will still try to manage the database.  

Docs: https://github.com/ansible/awx-operator/#external-postgresql-service

**Comment by AlekseiSaff** [comment]
That's correct. So I created new secret for externalDB with
`---
apiVersion: v1
kind: Secret
metadata:
  name: awx-external-postgres-configuration
  namespace: default
stringData:
  host: "awx"
  port: "5432"
  database: "awx"
  username: "awx"
  password: "xxxxxxx"
  sslmode: allow
  type: unmanaged
type: Opaque
`

and applied new awx template with the sting
  postgres_configuration_secret: awx-external-postgres-configuration

and my kubectl is still showing postgres is running



[lt@hostname AWX]$ kubectl get pods
NAME                                               READY   STATUS    RESTARTS   AGE
awx-d75477f8d-6l9b5                                4/4     Running   8          8d
awx-operator-controller-manager-6785cbd4c5-xsw4v   2/2     Running   4          16d
awx-postgres-0                                     1/1     Running   1          8d

**Comment by AlekseiSaff** [comment]
my AWX works fine and I can see it's using new external DB

**Comment by kurokobo** [comment]
@AlekseiSaff 
If you've created AWX once with `managed` and then changed it to `unmanaged`, AWX Operator will not delete the old PostgreSQL. 

So if your `awx-postgres-0` is no longer needed, I would suggest simply removing it. Since this Pod is created by StatefulSet, it is the StatefulSet that should be deleted.

```bash
kubectl -n <namespace> delete statefulset awx-postgres
```

Operator tries to keep objects that should be managed by Operator in desired state but, the Operator is not involved with objects that are not under the Operator's control. This is a designed behavior of Kubernetes Operator.

In other words, when you change to unmanaged, your AWX Operator will start maintaining an AWX with unmanaged db and will have forgotten about old PostgreSQL that was managed state before.

**Comment by AlekseiSaff** [comment]
makes sense. thanks!

**Comment by AshokMishra** [comment]
I am trying to use external database but whenever I try to perform necessary changes mentioned at - https://github.com/ansible/awx-operator#external-postgresql-service the awx deployment pods gets stuck in migration loop and never recovers.

```
[wait-for-migrations] Waiting for database migrations...
[wait-for-migrations] Attempt 1 of 30
[wait-for-migrations] Waiting 0.5 seconds before next attempt
[wait-for-migrations] Attempt 2 of 30
[wait-for-migrations] Waiting 1 seconds before next attempt
[wait-for-migrations] Attempt 3 of 30
[wait-for-migrations] Waiting 2 seconds before next attempt
[wait-for-migrations] Attempt 4 of 30
[wait-for-migrations] Waiting 4 seconds before next attempt
[wait-for-migrations] Attempt 5 of 30
[wait-for-migrations] Waiting 8 seconds before next attempt
[wait-for-migrations] Attempt 6 of 30
[wait-for-migrations] Waiting 16 seconds before next attempt
[wait-for-migrations] Attempt 7 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
```

**Comment by rooftopcellist** [comment]
It looks like @AlekseiSaff 's original issue has been resolved.  It is intended that we leave behind the old managed postgresql pod in that scenario in case the user wants to fall back to it.  We generally stay on the safe side and don't delete user data when possible.  

As mentioned earlier, you can delete the postgres pod manually.  We should add a note about this in the docs though.  

@AshokMishra this seems unrelated to the original issue.  However, I just deployed a fresh awx instance with an external postgres instance configured and it came up as expected.  The reconciliation loop converged and the migrations completed.

![image](https://user-images.githubusercontent.com/11698892/182866107-c3475b0b-6dac-4892-a834-c3e2a4ac4f32.png)


It is often useful to check how far along the migrations are by running `awx-manage showmigrations` inside the awx-task container.  

```
$ oc exec -it awx-554ccdb677-jlctz -c awx-task bash
bash-5.1$ awx-manage showmigrations
```

It does take a few minutes for the migrations to complete.  Also, make sure that if you are configuring an external database, your pg config secret has `type: unmanaged` in it.  

@AshokMishra If you are still experiencing an issue, please open a separate issue.
